packages/trader/src/AppV2/Containers/Trade/trade.tsx

```diff
@@ -1,13 +1,191 @@
 import React from 'react';
-import { Text } from '@deriv-com/quill-ui';
+import { ActionSheet } from '@deriv-com/quill-ui';
+import { observer } from '@deriv/stores';
 import BottomNav from 'AppV2/Components/BottomNav';
+import { DraggableList } from 'AppV2/Components/DraggableList';
+import { TradeTypeList } from 'AppV2/Components/TradeTypeList';
+import { useTraderStore } from 'Stores/useTraderStores';
+import { Localize } from '@deriv/translations';
+
+const Trade = observer(() => {
+    const trade_store = useTraderStore();
+    const { contract_types_list, prepareTradeStore } = trade_store;
+
+    type Category = {
+        text: string;
+        value: string;
+    };
+
+    type Item = {
+        id: string;
+        title: string;
+        icon?: React.ReactNode;
+    };
+
+    type DataObject = {
+        [key: string]: {
+            categories: Category[];
+        };
+    };
+
+    type ResultItem = {
+        id: string;
+        title?: string;
+        button_title?: string;
+        onButtonClick?: () => void;
+        items: Item[];
+    };
+
+    const createArrayFromCategories = (data: DataObject): Item[] => {
+        const result: Item[] = [];
+        let id_counter = 1;
+
+        for (const key in data) {
+            if (data.hasOwnProperty(key) && data[key].categories) {
+                data[key].categories.forEach(category => {
+                    result.push({
+                        id: id_counter.toString(),
+                        title: category.text,
+                    });
+                    id_counter++;
+                });
+            }
+        }
+
+        return result;
+    };
+
+    const [other_trade_types, setOtherTradeTypes] = React.useState<ResultItem[]>([]);
+    const [pinned_trade_types, setPinnedTradeTypes] = React.useState<ResultItem[]>([]);
+
+    const [open, setOpen] = React.useState<boolean>();
+
+    const handleOpenTradeTypes = () => {
+        const saved_other_trade_types = localStorage.getItem('other_trade_types');
+        const saved_pinned_trade_types = localStorage.getItem('pinned_trade_types');
+
+        if (saved_other_trade_types) {
+            setOtherTradeTypes(JSON.parse(saved_other_trade_types));
+        }
+
+        if (saved_pinned_trade_types) {
+            setPinnedTradeTypes(JSON.parse(saved_pinned_trade_types));
+        }
+
+        setOpen(true);
+    };
+
+    const handleAddPinnedClick = (item: Item) => {
+        setOtherTradeTypes(prev_categories =>
+            prev_categories.map(category => ({
+                ...category,
```

Comment:
I think spread operator doesn't deep cloning, only shallow. If you want deep copy, then we have `cloneObject` from shared or global` structuredClone()` method (but if I remember correctly it's not working with `React.ElementType`

Reply:
Ignore if irrelevant

Reply:
Hmm... It doesn't really matter, as long as it works)